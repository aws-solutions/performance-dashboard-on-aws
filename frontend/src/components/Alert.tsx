/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";

interface Props {
    title?: string;
    hideIcon?: boolean;
    message: string | React.ReactNode;
    type: "info" | "warning" | "error" | "success";
    slim?: boolean;
}

function Alert(props: Props) {
    let className = "usa-alert usa-alert--".concat(props.type);
    if (props.slim) {
        className += " usa-alert--slim";
    }

    if (props.hideIcon) {
        className += " usa-alert--no-icon";
    }

    return (
        <div className={className} role="alert">
            {typeof props.message === "object" ? (
                props.message
            ) : (
                <div className="usa-alert__body">
                    {props.title && !props.slim && (
                        <h2 className="usa-alert__heading">{props.title}</h2>
                    )}
                    <p className="usa-alert__text">{props.message}</p>
                </div>
            )}
        </div>
    );
}

export default Alert;
