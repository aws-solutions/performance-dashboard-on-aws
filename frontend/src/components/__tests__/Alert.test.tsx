/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import Alert from "../Alert";

test("renders a info alert", async () => {
    const wrapper = render(<Alert type="info" message="Hello world" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a warning alert", async () => {
    const wrapper = render(<Alert type="warning" message="Hello world" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders an error alert", async () => {
    const wrapper = render(<Alert type="error" message="Hello world" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a success alert", async () => {
    const wrapper = render(<Alert type="success" message="Hello world" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders an alert with title", async () => {
    const wrapper = render(
        <Alert type="success" message="Hello world" title="Informative status" />,
    );
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a slim alert", async () => {
    const wrapper = render(<Alert type="success" message="Hello world" slim />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a slim alert ignoring title", async () => {
    const wrapper = render(
        <Alert type="success" message="Hello world" slim title="This should be ignored" />,
    );
    expect(wrapper.container).toMatchSnapshot();
});

test("renders an alert without icon", async () => {
    const wrapper = render(<Alert type="success" message="Hello world" hideIcon slim />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders an alert with HTML content as message", async () => {
    const wrapper = render(
        <Alert type="success" message={<a href="/">This is a link</a>} hideIcon slim />,
    );
    expect(wrapper.container).toMatchSnapshot();
});
