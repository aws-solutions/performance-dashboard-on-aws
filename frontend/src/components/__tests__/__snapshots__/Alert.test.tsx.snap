// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a info alert 1`] = `
<div>
  <div
    class="usa-alert usa-alert--info"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a slim alert 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success usa-alert--slim"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a slim alert ignoring title 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success usa-alert--slim"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a success alert 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a warning alert 1`] = `
<div>
  <div
    class="usa-alert usa-alert--warning"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders an alert with HTML content as message 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success usa-alert--slim usa-alert--no-icon"
    role="alert"
  >
    <a
      href="/"
    >
      This is a link
    </a>
  </div>
</div>
`;

exports[`renders an alert with title 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <h2
        class="usa-alert__heading"
      >
        Informative status
      </h2>
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders an alert without icon 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success usa-alert--slim usa-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;

exports[`renders an error alert 1`] = `
<div>
  <div
    class="usa-alert usa-alert--error"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        Hello world
      </p>
    </div>
  </div>
</div>
`;
